query-service/src/integrationTest/java/org/hypertrace/core/query/service/QueryServiceTestUtils.java

```diff
@@ -33,40 +35,69 @@ public static Filter createFilter(
+    return Expression.newBuilder()
+        .setFunction(
+            Function.newBuilder()
+                .setFunctionName("dateTimeConvert")
+                .addArguments(
+                    Expression.newBuilder()
+                        .setColumnIdentifier(
+                            ColumnIdentifier.newBuilder().setColumnName(timeColumn)))
+                .addArguments(
+                    Expression.newBuilder()
+                        .setLiteral(
+                            LiteralConstant.newBuilder()
+                                .setValue(Value.newBuilder().setString("1:MILLISECONDS:EPOCH"))))
+                .addArguments(
+                    Expression.newBuilder()
+                        .setLiteral(
+                            LiteralConstant.newBuilder()
+                                .setValue(Value.newBuilder().setString("1:MILLISECONDS:EPOCH"))))
+                .addArguments(
+                    Expression.newBuilder()
+                        .setLiteral(
+                            LiteralConstant.newBuilder()
+                                .setValue(Value.newBuilder().setString(periodSecs))))
+                .build())
+        .build();
+  }
+
+  public static Expression createColumnExpression(String columnName) {
+    return Expression.newBuilder()
+        .setColumnIdentifier(ColumnIdentifier.newBuilder().setColumnName(columnName))
+        .build();
   }
 
-  public static ColumnIdentifier createColumnIdentifier(String columnName) {
-    return ColumnIdentifier.newBuilder().setColumnName(columnName).build();
+  public static Expression createFunctionExpression(String functionName, Expression columnNameArg) {
+    return Expression.newBuilder()
+        .setFunction(
+            Function.newBuilder().setFunctionName(functionName).addArguments(columnNameArg))
+        .build();
   }
 
-  public static Expression.Builder createColumnExpression(String columnName) {
+  public static Expression createFunctionExpression(
+      String functionName, Expression columnNameArg1, Expression columnNameArg2) {
     return Expression.newBuilder()
-        .setColumnIdentifier(ColumnIdentifier.newBuilder().setColumnName(columnName));
+        .setFunction(
+            Function.newBuilder()
+                .setFunctionName(functionName)
+                .addArguments(columnNameArg1)
+                .addArguments(columnNameArg2))
+        .build();
+  }
+
+  public static OrderByExpression createOrderByExpression(
+      Expression expression, SortOrder sortOrder) {
+    return OrderByExpression.newBuilder().setExpression(expression).setOrder(sortOrder).build();
   }
 
-  public static ColumnIdentifier createColumnIdentifier(String columnName, String alias) {
-    return ColumnIdentifier.newBuilder().setColumnName(columnName).setAlias(alias).build();
+  public static Expression createStringLiteralValueExpression(String value) {
```

Comment:
This exists

Reply:
```diff
diff --git a/query-service/src/integrationTest/java/org/hypertrace/core/query/service/QueryServiceTestUtils.java b/query-service/src/integrationTest/java/org/hypertrace/core/query/service/QueryServiceTestUtils.java
index 7729034..9e7c6e4 100644
--- a/query-service/src/integrationTest/java/org/hypertrace/core/query/service/QueryServiceTestUtils.java
+++ b/query-service/src/integrationTest/java/org/hypertrace/core/query/service/QueryServiceTestUtils.java
@@ -64,12 +64,6 @@ public class QueryServiceTestUtils {
         .build();
   }
 
-  public static Expression createColumnExpression(String columnName) {
-    return Expression.newBuilder()
-        .setColumnIdentifier(ColumnIdentifier.newBuilder().setColumnName(columnName))
-        .build();
-  }
-
   public static Expression createFunctionExpression(
       String functionName, Expression argumentExpression) {
     return Expression.newBuilder()
@@ -93,12 +87,4 @@ public class QueryServiceTestUtils {
       Expression expression, SortOrder sortOrder) {
     return OrderByExpression.newBuilder().setExpression(expression).setOrder(sortOrder).build();
   }
-
-  public static Expression createStringLiteralValueExpression(String value) {
-    return Expression.newBuilder()
-        .setLiteral(
-            LiteralConstant.newBuilder()
-                .setValue(Value.newBuilder().setString(value).setValueType(ValueType.STRING)))
-        .build();
-  }
 }
diff --git a/query-service/src/integrationTest/java/org/hypertrace/core/query/service/htqueries/ServicesQueries.java b/query-service/src/integrationTest/java/org/hypertrace/core/query/service/htqueries/ServicesQueries.java
index e29ede4..b2b2e10 100644
--- a/query-service/src/integrationTest/java/org/hypertrace/core/query/service/htqueries/ServicesQueries.java
+++ b/query-service/src/integrationTest/java/org/hypertrace/core/query/service/htqueries/ServicesQueries.java
@@ -1,10 +1,10 @@
 package org.hypertrace.core.query.service.htqueries;
 
-import static org.hypertrace.core.query.service.QueryServiceTestUtils.createColumnExpression;
+import static org.hypertrace.core.query.service.QueryRequestUtil.createColumnExpression;
+import static org.hypertrace.core.query.service.QueryRequestUtil.createStringLiteralExpression;
 import static org.hypertrace.core.query.service.QueryServiceTestUtils.createFilter;
 import static org.hypertrace.core.query.service.QueryServiceTestUtils.createFunctionExpression;
 import static org.hypertrace.core.query.service.QueryServiceTestUtils.createOrderByExpression;
-import static org.hypertrace.core.query.service.QueryServiceTestUtils.createStringLiteralValueExpression;
 import static org.hypertrace.core.query.service.QueryServiceTestUtils.createTimeColumnGroupByFunctionExpression;
 
 import java.time.Duration;
@@ -79,7 +79,7 @@ class ServicesQueries {
     Expression durationSumFunction = createFunctionExpression("SUM", serviceDuration);
     Expression durationAvgRateFunction =
         createFunctionExpression(
-            "AVG_RATE", serviceDuration, createStringLiteralValueExpression("PT1S"));
+            "AVG_RATE", serviceDuration, createStringLiteralExpression("PT1S"));
 
     builder.addSelection(durationAvgRateFunction);
     builder.addSelection(durationSumFunction);
@@ -123,7 +123,7 @@ class ServicesQueries {
     Expression serviceNumCalls = createColumnExpression("SERVICE.numCalls");
     Expression durationAvgRateFunction =
         createFunctionExpression(
-            "AVG_RATE", serviceNumCalls, createStringLiteralValueExpression("PT1S"));
+            "AVG_RATE", serviceNumCalls, createStringLiteralExpression("PT1S"));
     Expression durationSumFunction = createFunctionExpression("SUM", serviceNumCalls);
 
     builder.addSelection(durationAvgRateFunction);
```

compiles fine - integration tests have access to the implementation they're testing, they just don't have access to the unit tests.